bioptim/limits/penalty.py

```diff
@@ -536,7 +537,10 @@ def continuity(penalty: PenaltyOption, all_pn: Union[PenaltyNodeList, list]):
                 continuity = vertcat(continuity, nlp.dynamics[0](x0=cx, p=u, params=nlp.parameters.cx)["defects"])
                 penalty.integrate = True
             else:
-                continuity -= nlp.dynamics[0](x0=nlp.states.cx, p=u, params=nlp.parameters.cx)["xf"]
+                if isinstance(nlp.ode_solver, OdeSolver.CVODES):
+                    continuity -= nlp.dynamics[0](x0=nlp.states.cx, p=u)["xf"]
```

Comment:
That is an unsatisfying method
nlp.dynamics{0] should be wrapped in a casadi.Function welcoming the `params` parameter (even though it simply ignores it). Find where nlp.dynamics[0] (and other indices) is set and wrap it in a function

Reply:
Once it is done, this if instance can fully be removed